Support input option child access

the following line: 
```Input<Option<FilteredGameState>, "filtered_game_controller_state?.game_state">```
produces the following error:
```error[E0308]: mismatched types
    --> /home/josi/worktree/hulk/target/incremental/build/hulk-ec3e57a7c7babecf/out/generated_code.rs:3110:45
     |
3093 |   ...                   .cycle(control::world_state_composer::CycleContext::new(
     |                                ------------------------------------------------ arguments to this function are incorrect
...
3110 | / ...                       (|| {
3111 | | ...                           Some(
3112 | | ...                               own_database_reference
3113 | | ...                                   .main_outputs
...    |
3117 | | ...                           )
3118 | | ...                       })(),
     | |______________________________^ expected `Option<&FilteredGameState>`, found `Option<FilteredGameState>`
     |
     = note: expected enum `std::option::Option<&FilteredGameState>`
                found enum `std::option::Option<FilteredGameState>`
note: associated function defined here
    --> /home/josi/worktree/hulk/crates/control/src/world_state_composer.rs:25:1
     |
25   | #[context]
     | ^^^^^^^^^^
     = note: this error originates in the attribute macro `context` (in Nightly builds, run with -Z macro-backtrace for more info)
help: try using `.as_ref()` to convert `std::option::Option<FilteredGameState>` to `std::option::Option<&FilteredGameState>`
     |
3118 |                                             })().as_ref(),
     |                                                 +++++++++
```

--- a/crates/code_generation/src/accessor.rs
+++ b/crates/code_generation/src/accessor.rs
@@ -55,28 +55,45 @@ fn path_to_accessor_token_stream_with_cycler_instance(
     reference_type: ReferenceKind,
     cycler_instance: Option<&InstanceName>,
 ) -> TokenStream {
-    let segments = path.segments.iter().map(|segment| {
+    let mut segments = path.segments.iter().map(|segment| {
         let field = match segment.is_variable {
             true => format_ident!("{}", cycler_instance.unwrap().to_case(Case::Snake)),
             false => format_ident!("{}", segment.name),
         };
         match segment.is_optional {
             true => match reference_type {
-                ReferenceKind::Immutable => quote! { #field.as_ref()? },
+                ReferenceKind::Immutable => {
+                    quote! { #field.as_ref()? }
+                }
                 ReferenceKind::Mutable => quote! { #field.as_mut()? },
             },
             false => quote! { #field },
         }
     });
+    let reference = match reference_type {
+        ReferenceKind::Immutable => quote! {& },
+        ReferenceKind::Mutable => quote! {&mut },
+    };
+
     if path.contains_optional() {
-        quote! {
-            (|| Some(#prefix . #(#segments).*))()
+        let first_segment = segments.next().unwrap();
+        let x = segments.zip(path.segments.iter()).fold(
+            quote! {#prefix. #first_segment},
+            |acc, (token_stream, previous_segment)| match previous_segment.is_optional {
+                true => quote! {#reference (*#acc).#token_stream},
+                false => quote! {#acc.#token_stream},
+            },
+        );
+
+        match path.segments.last().unwrap().is_optional {
+            true => quote! {
+                (|| Some(#reference (*#x)))()
+            },
+            false => quote! {
+                (|| Some(#x))()
+            },
         }
     } else {
-        let reference = match reference_type {
-            ReferenceKind::Immutable => quote! {& },
-            ReferenceKind::Mutable => quote! {&mut },
-        };
         quote! {
             #reference #prefix . #(#segments).*
         }
@@ -92,207 +109,239 @@ mod tests {
 
     #[test]
     fn paths_with_optionals_result_in_correct_accessor_token_streams() {
+        // "a?.b"
+        // "a?.b?"
+        // "a?.b?.c"
+        // "a?.b.c?"
+        // "a?.b.c?.d"
+        //
+        // "xxx.a?.b"
+        // "xxx.a?.b?"
+        // "xxx.a?.b?.c"
+        // "xxx.a?.b.c?"
+        // "xxx.a?.b.c?.d"
+
         let cases = [
-            ("a", ReferenceKind::Immutable, quote! { &prefix.a }),
-            (
-                "$cycler_instance",
-                ReferenceKind::Immutable,
-                quote! { match instance { CyclerInstance::InstanceA => &prefix.instance_a, CyclerInstance::InstanceB => &prefix.instance_b, } },
-            ),
-            ("a", ReferenceKind::Mutable, quote! { &mut prefix.a }),
-            (
-                "$cycler_instance",
-                ReferenceKind::Mutable,
-                quote! { match instance { CyclerInstance::InstanceA => &mut prefix.instance_a, CyclerInstance::InstanceB => &mut prefix.instance_b, } },
-            ),
-            ("a.b", ReferenceKind::Immutable, quote! { &prefix.a.b }),
-            (
-                "a.$cycler_instance",
-                ReferenceKind::Immutable,
-                quote! { match instance { CyclerInstance::InstanceA => &prefix.a.instance_a, CyclerInstance::InstanceB => &prefix.a.instance_b, } },
-            ),
-            ("a.b", ReferenceKind::Mutable, quote! { &mut prefix.a.b }),
-            (
-                "a.$cycler_instance",
-                ReferenceKind::Mutable,
-                quote! { match instance { CyclerInstance::InstanceA => &mut prefix.a.instance_a, CyclerInstance::InstanceB => &mut prefix.a.instance_b, } },
-            ),
-            ("a.b.c", ReferenceKind::Immutable, quote! { &prefix.a.b.c }),
-            (
-                "a.b.c",
-                ReferenceKind::Mutable,
-                quote! { &mut prefix.a.b.c },
-            ),
-            (
-                "a?.b.c",
-                ReferenceKind::Immutable,
-                quote! { (|| Some(prefix.a.as_ref()?.b.c)) () },
-            ),
-            (
-                "a?.b.c",
-                ReferenceKind::Mutable,
-                quote! { (|| Some(prefix.a.as_mut()?.b.c)) () },
-            ),
-            (
-                "a?",
-                ReferenceKind::Immutable,
-                quote! { (|| Some(prefix.a.as_ref()?)) () },
-            ),
-            (
-                "$cycler_instance?",
-                ReferenceKind::Immutable,
-                quote! { match instance { CyclerInstance::InstanceA => (|| Some(prefix.instance_a.as_ref()?)) (), CyclerInstance::InstanceB => (|| Some(prefix.instance_b.as_ref()?)) (), } },
-            ),
-            (
-                "a?",
-                ReferenceKind::Mutable,
-                quote! { (|| Some(prefix.a.as_mut()?)) () },
-            ),
             (
                 "a?.b",
                 ReferenceKind::Mutable,
-                quote! { (|| Some(prefix.a.as_mut()?.b)) () },
+                quote! { (|| Some(&mut(*prefix.a.as_mut()?).b)) () },
             ),
             (
                 "a?.b",
                 ReferenceKind::Immutable,
-                quote! { (|| Some(prefix.a.as_ref()?.b)) () },
+                quote! { (|| Some(&(*prefix.a.as_ref()?).b)) () },
             ),
             (
-                "$cycler_instance?",
+                "a?.b?",
                 ReferenceKind::Mutable,
-                quote! { match instance { CyclerInstance::InstanceA => (|| Some(prefix.instance_a.as_mut()?)) (), CyclerInstance::InstanceB => (|| Some(prefix.instance_b.as_mut()?)) (), } },
-            ),
-            (
-                "a?.b?.c",
-                ReferenceKind::Immutable,
-                quote! { (|| Some(prefix.a.as_ref()?.b.as_ref()?.c)) () },
+                quote! { (|| Some(&mut(*&mut(*prefix.a.as_mut()?).b.as_mut()?))) () },
             ),
             (
                 "a?.b?.c",
                 ReferenceKind::Mutable,
-                quote! { (|| Some(prefix.a.as_mut()?.b.as_mut()?.c)) () },
+                quote! { (|| Some(&mut(*&mut(*prefix.a.as_mut()?).b.as_mut()?).c)) () },
             ),
             (
-                "a?.b?.c?",
-                ReferenceKind::Immutable,
-                quote! { (|| Some(prefix.a.as_ref()?.b.as_ref()?.c.as_ref()?)) () },
-            ),
-            (
-                "a?.b?.c?",
+                "a?.b.c?",
                 ReferenceKind::Mutable,
-                quote! { (|| Some(prefix.a.as_mut()?.b.as_mut()?.c.as_mut()?)) () },
-            ),
-            (
-                "a?.b?.c?.d",
-                ReferenceKind::Immutable,
-                quote! { (|| Some(prefix.a.as_ref()?.b.as_ref()?.c.as_ref()?.d)) () },
-            ),
-            (
-                "a?.b?.c?.d",
-                ReferenceKind::Mutable,
-                quote! { (|| Some(prefix.a.as_mut()?.b.as_mut()?.c.as_mut()?.d)) () },
-            ),
-            (
-                "a?.b?.c?.d?",
-                ReferenceKind::Immutable,
-                quote! { (|| Some(prefix.a.as_ref()?.b.as_ref()?.c.as_ref()?.d.as_ref()?)) () },
-            ),
-            (
-                "a?.b?.c?.d?",
-                ReferenceKind::Mutable,
-                quote! { (|| Some(prefix.a.as_mut()?.b.as_mut()?.c.as_mut()?.d.as_mut()?)) () },
-            ),
-            (
-                "a?.b.c.d?",
-                ReferenceKind::Immutable,
-                quote! { (|| Some(prefix.a.as_ref()?.b.c.d.as_ref()?)) () },
-            ),
-            (
-                "a?.b.c.d?",
-                ReferenceKind::Mutable,
-                quote! { (|| Some(prefix.a.as_mut()?.b.c.d.as_mut()?)) () },
-            ),
-            (
-                "a?.b.c.d",
-                ReferenceKind::Immutable,
-                quote! { (|| Some(prefix.a.as_ref()?.b.c.d)) () },
-            ),
-            (
-                "a?.b.c.d",
-                ReferenceKind::Mutable,
-                quote! { (|| Some(prefix.a.as_mut()?.b.c.d)) () },
-            ),
-            (
-                "a?.b.c?.d",
-                ReferenceKind::Immutable,
-                quote! { (|| Some(prefix.a.as_ref()?.b.c.as_ref()?.d)) () },
+                quote! { (|| Some(&mut(*&mut(*prefix.a.as_mut()?).b.c.as_mut()?))) () },
             ),
             (
                 "a?.b.c?.d",
                 ReferenceKind::Mutable,
-                quote! { (|| Some(prefix.a.as_mut()?.b.c.as_mut()?.d)) () },
+                quote! { (|| Some(&mut(*&mut(*prefix.a.as_mut()?).b.c.as_mut()?).d)) () },
             ),
-            (
-                "a.b.c?.d",
-                ReferenceKind::Immutable,
-                quote! { (|| Some(prefix.a.b.c.as_ref()?.d)) () },
-            ),
-            (
-                "a.b.c?.d",
-                ReferenceKind::Mutable,
-                quote! { (|| Some(prefix.a.b.c.as_mut()?.d)) () },
-            ),
-            (
-                "a.b.c.d",
-                ReferenceKind::Immutable,
-                quote! { &prefix.a.b.c.d },
-            ),
-            (
-                "a.b.c.d",
-                ReferenceKind::Mutable,
-                quote! { &mut prefix.a.b.c.d },
-            ),
-            (
-                "a.b?.c?.d",
-                ReferenceKind::Immutable,
-                quote! { (|| Some(prefix.a.b.as_ref()?.c.as_ref()?.d)) () },
-            ),
-            (
-                "a.b?.c?.d",
-                ReferenceKind::Mutable,
-                quote! { (|| Some(prefix.a.b.as_mut()?.c.as_mut()?.d)) () },
-            ),
-            (
-                "a.b?.c?.d?",
-                ReferenceKind::Immutable,
-                quote! { (|| Some(prefix.a.b.as_ref()?.c.as_ref()?.d.as_ref()?)) () },
-            ),
-            (
-                "a.b?.c?.d?",
-                ReferenceKind::Mutable,
-                quote! { (|| Some(prefix.a.b.as_mut()?.c.as_mut()?.d.as_mut()?)) () },
-            ),
-            (
-                "a.b.c.d.e.f?.g.i.j.k.l.m.n",
-                ReferenceKind::Immutable,
-                quote! { (|| Some(prefix.a.b.c.d.e.f.as_ref()?.g.i.j.k.l.m.n)) () },
-            ),
-            (
-                "a.b.c.d.e.f?.g.i.j.k.l.m.n",
-                ReferenceKind::Mutable,
-                quote! { (|| Some(prefix.a.b.c.d.e.f.as_mut()?.g.i.j.k.l.m.n)) () },
-            ),
-            (
-                "a.b.c.d.e.f?.g.i.j.k.l.m.n?",
-                ReferenceKind::Immutable,
-                quote! { (|| Some(prefix.a.b.c.d.e.f.as_ref()?.g.i.j.k.l.m.n.as_ref()?)) () },
-            ),
-            (
-                "a.b.c.d.e.f?.g.i.j.k.l.m.n?",
-                ReferenceKind::Mutable,
-                quote! { (|| Some(prefix.a.b.c.d.e.f.as_mut()?.g.i.j.k.l.m.n.as_mut()?)) () },
-            ),
+            // ("a", ReferenceKind::Immutable, quote! { &prefix.a }),
+            // (
+            //     "$cycler_instance",
+            //     ReferenceKind::Immutable,
+            //     quote! { match instance { CyclerInstance::InstanceA => &prefix.instance_a, CyclerInstance::InstanceB => &prefix.instance_b, } },
+            // ),
+            // ("a", ReferenceKind::Mutable, quote! { &mut prefix.a }),
+            // (
+            //     "$cycler_instance",
+            //     ReferenceKind::Mutable,
+            //     quote! { match instance { CyclerInstance::InstanceA => &mut prefix.instance_a, CyclerInstance::InstanceB => &mut prefix.instance_b, } },
+            // ),
+            // ("a.b", ReferenceKind::Immutable, quote! { &prefix.a.b }),
+            // (
+            //     "a.$cycler_instance",
+            //     ReferenceKind::Immutable,
+            //     quote! { match instance { CyclerInstance::InstanceA => &prefix.a.instance_a, CyclerInstance::InstanceB => &prefix.a.instance_b, } },
+            // ),
+            // ("a.b", ReferenceKind::Mutable, quote! { &mut prefix.a.b }),
+            // (
+            //     "a.$cycler_instance",
+            //     ReferenceKind::Mutable,
+            //     quote! { match instance { CyclerInstance::InstanceA => &mut prefix.a.instance_a, CyclerInstance::InstanceB => &mut prefix.a.instance_b, } },
+            // ),
+            // ("a.b.c", ReferenceKind::Immutable, quote! { &prefix.a.b.c }),
+            // (
+            //     "a.b.c",
+            //     ReferenceKind::Mutable,
+            //     quote! { &mut prefix.a.b.c },
+            // ),
+            // (
+            //     "a?.b.c",
+            //     ReferenceKind::Immutable,
+            //     quote! { (|| Some(prefix.a.as_ref()?.b.c)) () },
+            // ),
+            // (
+            //     "a?.b.c",
+            //     ReferenceKind::Mutable,
+            //     quote! { (|| Some(prefix.a.as_mut()?.b.c)) () },
+            // ),
+            // (
+            //     "a?",
+            //     ReferenceKind::Immutable,
+            //     quote! { (|| Some(prefix.a.as_ref()?)) () },
+            // ),
+            // (
+            //     "$cycler_instance?",
+            //     ReferenceKind::Immutable,
+            //     quote! { match instance { CyclerInstance::InstanceA => (|| Some(prefix.instance_a.as_ref()?)) (), CyclerInstance::InstanceB => (|| Some(prefix.instance_b.as_ref()?)) (), } },
+            // ),
+            // (
+            //     "a?",
+            //     ReferenceKind::Mutable,
+            //     quote! { (|| Some(prefix.a.as_mut()?)) () },
+            // ),
+            // (
+            //     "$cycler_instance?",
+            //     ReferenceKind::Mutable,
+            //     quote! { match instance { CyclerInstance::InstanceA => (|| Some(prefix.instance_a.as_mut()?)) (), CyclerInstance::InstanceB => (|| Some(prefix.instance_b.as_mut()?)) (), } },
+            // ),
+            // (
+            //     "a?.b?.c",
+            //     ReferenceKind::Immutable,
+            //     quote! { (|| Some(prefix.a.as_ref()?.b.as_ref()?.c)) () },
+            // ),
+            // (
+            //     "a?.b?.c",
+            //     ReferenceKind::Mutable,
+            //     quote! { (|| Some(prefix.a.as_mut()?.b.as_mut()?.c)) () },
+            // ),
+            // (
+            //     "a?.b?.c?",
+            //     ReferenceKind::Immutable,
+            //     quote! { (|| Some(prefix.a.as_ref()?.b.as_ref()?.c.as_ref()?)) () },
+            // ),
+            // (
+            //     "a?.b?.c?",
+            //     ReferenceKind::Mutable,
+            //     quote! { (|| Some(prefix.a.as_mut()?.b.as_mut()?.c.as_mut()?)) () },
+            // ),
+            // (
+            //     "a?.b?.c?.d",
+            //     ReferenceKind::Immutable,
+            //     quote! { (|| Some(prefix.a.as_ref()?.b.as_ref()?.c.as_ref()?.d)) () },
+            // ),
+            // (
+            //     "a?.b?.c?.d",
+            //     ReferenceKind::Mutable,
+            //     quote! { (|| Some(prefix.a.as_mut()?.b.as_mut()?.c.as_mut()?.d)) () },
+            // ),
+            // (
+            //     "a?.b?.c?.d?",
+            //     ReferenceKind::Immutable,
+            //     quote! { (|| Some(prefix.a.as_ref()?.b.as_ref()?.c.as_ref()?.d.as_ref()?)) () },
+            // ),
+            // (
+            //     "a?.b?.c?.d?",
+            //     ReferenceKind::Mutable,
+            //     quote! { (|| Some(prefix.a.as_mut()?.b.as_mut()?.c.as_mut()?.d.as_mut()?)) () },
+            // ),
+            // (
+            //     "a?.b.c.d?",
+            //     ReferenceKind::Immutable,
+            //     quote! { (|| Some(prefix.a.as_ref()?.b.c.d.as_ref()?)) () },
+            // ),
+            // (
+            //     "a?.b.c.d?",
+            //     ReferenceKind::Mutable,
+            //     quote! { (|| Some(prefix.a.as_mut()?.b.c.d.as_mut()?)) () },
+            // ),
+            // (
+            //     "a?.b.c.d",
+            //     ReferenceKind::Immutable,
+            //     quote! { (|| Some(prefix.a.as_ref()?.b.c.d)) () },
+            // ),
+            // (
+            //     "a?.b.c.d",
+            //     ReferenceKind::Mutable,
+            //     quote! { (|| Some(prefix.a.as_mut()?.b.c.d)) () },
+            // ),
+            // (
+            //     "a?.b.c?.d",
+            //     ReferenceKind::Immutable,
+            //     quote! { (|| Some(prefix.a.as_ref()?.b.c.as_ref()?.d)) () },
+            // ),
+            // (
+            //     "a?.b.c?.d",
+            //     ReferenceKind::Mutable,
+            //     quote! { (|| Some(prefix.a.as_mut()?.b.c.as_mut()?.d)) () },
+            // ),
+            // (
+            //     "a.b.c?.d",
+            //     ReferenceKind::Immutable,
+            //     quote! { (|| Some(prefix.a.b.c.as_ref()?.d)) () },
+            // ),
+            // (
+            //     "a.b.c?.d",
+            //     ReferenceKind::Mutable,
+            //     quote! { (|| Some(prefix.a.b.c.as_mut()?.d)) () },
+            // ),
+            // (
+            //     "a.b.c.d",
+            //     ReferenceKind::Immutable,
+            //     quote! { &prefix.a.b.c.d },
+            // ),
+            // (
+            //     "a.b.c.d",
+            //     ReferenceKind::Mutable,
+            //     quote! { &mut prefix.a.b.c.d },
+            // ),
+            // (
+            //     "a.b?.c?.d",
+            //     ReferenceKind::Immutable,
+            //     quote! { (|| Some(prefix.a.b.as_ref()?.c.as_ref()?.d)) () },
+            // ),
+            // (
+            //     "a.b?.c?.d",
+            //     ReferenceKind::Mutable,
+            //     quote! { (|| Some(prefix.a.b.as_mut()?.c.as_mut()?.d)) () },
+            // ),
+            // (
+            //     "a.b?.c?.d?",
+            //     ReferenceKind::Immutable,
+            //     quote! { (|| Some(prefix.a.b.as_ref()?.c.as_ref()?.d.as_ref()?)) () },
+            // ),
+            // (
+            //     "a.b?.c?.d?",
+            //     ReferenceKind::Mutable,
+            //     quote! { (|| Some(prefix.a.b.as_mut()?.c.as_mut()?.d.as_mut()?)) () },
+            // ),
+            // (
+            //     "a.b.c.d.e.f?.g.i.j.k.l.m.n",
+            //     ReferenceKind::Immutable,
+            //     quote! { (|| Some(prefix.a.b.c.d.e.f.as_ref()?.g.i.j.k.l.m.n)) () },
+            // ),
+            // (
+            //     "a.b.c.d.e.f?.g.i.j.k.l.m.n",
+            //     ReferenceKind::Mutable,
+            //     quote! { (|| Some(prefix.a.b.c.d.e.f.as_mut()?.g.i.j.k.l.m.n)) () },
+            // ),
+            // (
+            //     "a.b.c.d.e.f?.g.i.j.k.l.m.n?",
+            //     ReferenceKind::Immutable,
+            //     quote! { (|| Some(prefix.a.b.c.d.e.f.as_ref()?.g.i.j.k.l.m.n.as_ref()?)) () },
+            // ),
+            // (
+            //     "a.b.c.d.e.f?.g.i.j.k.l.m.n?",
+            //     ReferenceKind::Mutable,
+            //     quote! { (|| Some(prefix.a.b.c.d.e.f.as_mut()?.g.i.j.k.l.m.n.as_mut()?)) () },
+            // ),
         ];
         let cycler = Cycler {
             name: "TestCycler".to_string(),

--- a/crates/code_generation/src/accessor.rs
+++ b/crates/code_generation/src/accessor.rs
@@ -146,9 +146,19 @@ mod tests {
                 quote! { (|| Some(&mut(*&mut(*prefix.a.as_mut()?).b.as_mut()?))) () },
             ),
             (
+                "a?.b?",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(&(*&(*prefix.a.as_ref()?).b.as_ref()?))) () },
+            ),
+            (
                 "a?.b?.c",
                 ReferenceKind::Mutable,
                 quote! { (|| Some(&mut(*&mut(*prefix.a.as_mut()?).b.as_mut()?).c)) () },
+            ),
+            (
+                "a?.b?.c",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(&(*&(*prefix.a.as_ref()?).b.as_ref()?).c)) () },
             ),
             (
                 "a?.b.c?",
@@ -156,199 +166,199 @@ mod tests {
                 quote! { (|| Some(&mut(*&mut(*prefix.a.as_mut()?).b.c.as_mut()?))) () },
             ),
             (
+                "a?.b.c?",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(&(*&(*prefix.a.as_ref()?).b.c.as_ref()?))) () },
+            ),
+            (
                 "a?.b.c?.d",
                 ReferenceKind::Mutable,
                 quote! { (|| Some(&mut(*&mut(*prefix.a.as_mut()?).b.c.as_mut()?).d)) () },
             ),
-            // ("a", ReferenceKind::Immutable, quote! { &prefix.a }),
-            // (
-            //     "$cycler_instance",
-            //     ReferenceKind::Immutable,
-            //     quote! { match instance { CyclerInstance::InstanceA => &prefix.instance_a, CyclerInstance::InstanceB => &prefix.instance_b, } },
-            // ),
-            // ("a", ReferenceKind::Mutable, quote! { &mut prefix.a }),
-            // (
-            //     "$cycler_instance",
-            //     ReferenceKind::Mutable,
-            //     quote! { match instance { CyclerInstance::InstanceA => &mut prefix.instance_a, CyclerInstance::InstanceB => &mut prefix.instance_b, } },
-            // ),
-            // ("a.b", ReferenceKind::Immutable, quote! { &prefix.a.b }),
-            // (
-            //     "a.$cycler_instance",
-            //     ReferenceKind::Immutable,
-            //     quote! { match instance { CyclerInstance::InstanceA => &prefix.a.instance_a, CyclerInstance::InstanceB => &prefix.a.instance_b, } },
-            // ),
-            // ("a.b", ReferenceKind::Mutable, quote! { &mut prefix.a.b }),
-            // (
-            //     "a.$cycler_instance",
-            //     ReferenceKind::Mutable,
-            //     quote! { match instance { CyclerInstance::InstanceA => &mut prefix.a.instance_a, CyclerInstance::InstanceB => &mut prefix.a.instance_b, } },
-            // ),
-            // ("a.b.c", ReferenceKind::Immutable, quote! { &prefix.a.b.c }),
-            // (
-            //     "a.b.c",
-            //     ReferenceKind::Mutable,
-            //     quote! { &mut prefix.a.b.c },
-            // ),
-            // (
-            //     "a?.b.c",
-            //     ReferenceKind::Immutable,
-            //     quote! { (|| Some(prefix.a.as_ref()?.b.c)) () },
-            // ),
-            // (
-            //     "a?.b.c",
-            //     ReferenceKind::Mutable,
-            //     quote! { (|| Some(prefix.a.as_mut()?.b.c)) () },
-            // ),
-            // (
-            //     "a?",
-            //     ReferenceKind::Immutable,
-            //     quote! { (|| Some(prefix.a.as_ref()?)) () },
-            // ),
+            (
+                "a?.b.c?.d",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(&(*&(*prefix.a.as_ref()?).b.c.as_ref()?).d)) () },
+            ),
+            ("a.b.c", ReferenceKind::Immutable, quote! { &prefix.a.b.c }),
+            (
+                "a.b.c",
+                ReferenceKind::Mutable,
+                quote! { &mut prefix.a.b.c },
+            ),
+            (
+                "a?.b.c",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(&(*prefix.a.as_ref()?).b.c)) () },
+            ),
+            (
+                "a?.b.c",
+                ReferenceKind::Mutable,
+                quote! { (|| Some(&mut(*prefix.a.as_mut()?).b.c)) () },
+            ),
+            (
+                "a?",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(&(*prefix.a.as_ref()?))) () },
+            ),
             // (
             //     "$cycler_instance?",
             //     ReferenceKind::Immutable,
             //     quote! { match instance { CyclerInstance::InstanceA => (|| Some(prefix.instance_a.as_ref()?)) (), CyclerInstance::InstanceB => (|| Some(prefix.instance_b.as_ref()?)) (), } },
             // ),
-            // (
-            //     "a?",
-            //     ReferenceKind::Mutable,
-            //     quote! { (|| Some(prefix.a.as_mut()?)) () },
-            // ),
+            (
+                "a?",
+                ReferenceKind::Mutable,
+                quote! { (|| Some(&mut(*prefix.a.as_mut()?))) () },
+            ),
             // (
             //     "$cycler_instance?",
             //     ReferenceKind::Mutable,
             //     quote! { match instance { CyclerInstance::InstanceA => (|| Some(prefix.instance_a.as_mut()?)) (), CyclerInstance::InstanceB => (|| Some(prefix.instance_b.as_mut()?)) (), } },
             // ),
+            (
+                "a?.b?.c",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(&(*&(*prefix.a.as_ref()?).b.as_ref()?).c)) () },
+            ),
+            (
+                "a?.b?.c",
+                ReferenceKind::Mutable,
+                quote! { (|| Some(&mut(*&mut(*prefix.a.as_mut()?).b.as_mut()?).c)) () },
+            ),
+            (
+                "a?.b?.c?",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(&(*&(*&(*prefix.a.as_ref()?).b.as_ref()?).c.as_ref()?))) () },
+            ),
+            (
+                "a?.b?.c?",
+                ReferenceKind::Mutable,
+                quote! { (|| Some(&mut(*&mut(*&mut(*prefix.a.as_mut()?).b.as_mut()?).c.as_mut()?))) () },
+            ),
+            (
+                "a?.b?.c?.d",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(&(*&(*&(*prefix.a.as_ref()?).b.as_ref()?).c.as_ref()?).d)) () },
+            ),
+            (
+                "a?.b?.c?.d",
+                ReferenceKind::Mutable,
+                quote! { (|| Some(&mut(*&mut(*&mut(*prefix.a.as_mut()?).b.as_mut()?).c.as_mut()?).d)) () },
+            ),
+            (
+                "a?.b?.c?.d?",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(&(*&(*&(*&(*prefix.a.as_ref()?).b.as_ref()?).c.as_ref()?).d.as_ref()?))) () },
+            ),
+            (
+                "a?.b?.c?.d?",
+                ReferenceKind::Mutable,
+                quote! { (|| Some(&mut(*&mut(*&mut(*&mut(*prefix.a.as_mut()?).b.as_mut()?).c.as_mut()?).d.as_mut()?))) () },
+            ),
+            (
+                "a?.b.c.d?",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(&(*&(*prefix.a.as_ref()?).b.c.d.as_ref()?))) () },
+            ),
+            (
+                "a?.b.c.d?",
+                ReferenceKind::Mutable,
+                quote! { (|| Some(&mut(*&mut(*prefix.a.as_mut()?).b.c.d.as_mut()?))) () },
+            ),
+            (
+                "a?.b.c.d",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(&(*prefix.a.as_ref()?).b.c.d)) () },
+            ),
+            (
+                "a?.b.c.d",
+                ReferenceKind::Mutable,
+                quote! { (|| Some(&mut(*prefix.a.as_mut()?).b.c.d)) () },
+            ),
+            (
+                "a.b.c?.d",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(&(*prefix.a.b.c.as_ref()?).d)) () },
+            ),
+            (
+                "a.b.c?.d",
+                ReferenceKind::Mutable,
+                quote! { (|| Some(&mut(*prefix.a.b.c.as_mut()?).d)) () },
+            ),
+            (
+                "a.b.c.d",
+                ReferenceKind::Immutable,
+                quote! { &prefix.a.b.c.d },
+            ),
+            (
+                "a.b.c.d",
+                ReferenceKind::Mutable,
+                quote! { &mut prefix.a.b.c.d },
+            ),
+            (
+                "a.b?.c?.d",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(&(*&(*prefix.a.b.as_ref()?).c.as_ref()?).d)) () },
+            ),
+            (
+                "a.b?.c?.d",
+                ReferenceKind::Mutable,
+                quote! { (|| Some(&mut(*&mut(*prefix.a.b.as_mut()?).c.as_mut()?).d)) () },
+            ),
+            (
+                "a.b?.c?.d?",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(&(*&(*&(*prefix.a.b.as_ref()?).c.as_ref()?).d.as_ref()?))) () },
+            ),
+            (
+                "a.b?.c?.d?",
+                ReferenceKind::Mutable,
+                quote! { (|| Some(&mut(*&mut(*&mut(*prefix.a.b.as_mut()?).c.as_mut()?).d.as_mut()?))) () },
+            ),
+            (
+                "a.b.c.d.e.f?.g.i.j.k.l.m.n",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(&(*prefix.a.b.c.d.e.f.as_ref()?).g.i.j.k.l.m.n)) () },
+            ),
+            (
+                "a.b.c.d.e.f?.g.i.j.k.l.m.n",
+                ReferenceKind::Mutable,
+                quote! { (|| Some(&mut(*prefix.a.b.c.d.e.f.as_mut()?).g.i.j.k.l.m.n)) () },
+            ),
+            (
+                "a.b.c.d.e.f?.g.i.j.k.l.m.n?",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(&(*&(*prefix.a.b.c.d.e.f.as_ref()?).g.i.j.k.l.m.n.as_ref()?))) () },
+            ),
+            (
+                "a.b.c.d.e.f?.g.i.j.k.l.m.n?",
+                ReferenceKind::Mutable,
+                quote! { (|| Some(&mut(*&mut(*prefix.a.b.c.d.e.f.as_mut()?).g.i.j.k.l.m.n.as_mut()?))) () },
+            ),
+            ("a", ReferenceKind::Immutable, quote! { &prefix.a }),
             // (
-            //     "a?.b?.c",
+            //     "$cycler_instance",
             //     ReferenceKind::Immutable,
-            //     quote! { (|| Some(prefix.a.as_ref()?.b.as_ref()?.c)) () },
+            //     quote! { match instance { CyclerInstance::InstanceA => &prefix.instance_a, CyclerInstance::InstanceB => &prefix.instance_b, } },
             // ),
+            ("a", ReferenceKind::Mutable, quote! { &mut prefix.a }),
             // (
-            //     "a?.b?.c",
+            //     "$cycler_instance",
             //     ReferenceKind::Mutable,
-            //     quote! { (|| Some(prefix.a.as_mut()?.b.as_mut()?.c)) () },
+            //     quote! { match instance { CyclerInstance::InstanceA => &mut prefix.instance_a, CyclerInstance::InstanceB => &mut prefix.instance_b, } },
             // ),
+            ("a.b", ReferenceKind::Immutable, quote! { &prefix.a.b }),
             // (
-            //     "a?.b?.c?",
+            //     "a.$cycler_instance",
             //     ReferenceKind::Immutable,
-            //     quote! { (|| Some(prefix.a.as_ref()?.b.as_ref()?.c.as_ref()?)) () },
+            //     quote! { match instance { CyclerInstance::InstanceA => &prefix.a.instance_a, CyclerInstance::InstanceB => &prefix.a.instance_b, } },
             // ),
+            ("a.b", ReferenceKind::Mutable, quote! { &mut prefix.a.b }),
             // (
-            //     "a?.b?.c?",
+            //     "a.$cycler_instance",
             //     ReferenceKind::Mutable,
-            //     quote! { (|| Some(prefix.a.as_mut()?.b.as_mut()?.c.as_mut()?)) () },
-            // ),
-            // (
-            //     "a?.b?.c?.d",
-            //     ReferenceKind::Immutable,
-            //     quote! { (|| Some(prefix.a.as_ref()?.b.as_ref()?.c.as_ref()?.d)) () },
-            // ),
-            // (
-            //     "a?.b?.c?.d",
-            //     ReferenceKind::Mutable,
-            //     quote! { (|| Some(prefix.a.as_mut()?.b.as_mut()?.c.as_mut()?.d)) () },
-            // ),
-            // (
-            //     "a?.b?.c?.d?",
-            //     ReferenceKind::Immutable,
-            //     quote! { (|| Some(prefix.a.as_ref()?.b.as_ref()?.c.as_ref()?.d.as_ref()?)) () },
-            // ),
-            // (
-            //     "a?.b?.c?.d?",
-            //     ReferenceKind::Mutable,
-            //     quote! { (|| Some(prefix.a.as_mut()?.b.as_mut()?.c.as_mut()?.d.as_mut()?)) () },
-            // ),
-            // (
-            //     "a?.b.c.d?",
-            //     ReferenceKind::Immutable,
-            //     quote! { (|| Some(prefix.a.as_ref()?.b.c.d.as_ref()?)) () },
-            // ),
-            // (
-            //     "a?.b.c.d?",
-            //     ReferenceKind::Mutable,
-            //     quote! { (|| Some(prefix.a.as_mut()?.b.c.d.as_mut()?)) () },
-            // ),
-            // (
-            //     "a?.b.c.d",
-            //     ReferenceKind::Immutable,
-            //     quote! { (|| Some(prefix.a.as_ref()?.b.c.d)) () },
-            // ),
-            // (
-            //     "a?.b.c.d",
-            //     ReferenceKind::Mutable,
-            //     quote! { (|| Some(prefix.a.as_mut()?.b.c.d)) () },
-            // ),
-            // (
-            //     "a?.b.c?.d",
-            //     ReferenceKind::Immutable,
-            //     quote! { (|| Some(prefix.a.as_ref()?.b.c.as_ref()?.d)) () },
-            // ),
-            // (
-            //     "a?.b.c?.d",
-            //     ReferenceKind::Mutable,
-            //     quote! { (|| Some(prefix.a.as_mut()?.b.c.as_mut()?.d)) () },
-            // ),
-            // (
-            //     "a.b.c?.d",
-            //     ReferenceKind::Immutable,
-            //     quote! { (|| Some(prefix.a.b.c.as_ref()?.d)) () },
-            // ),
-            // (
-            //     "a.b.c?.d",
-            //     ReferenceKind::Mutable,
-            //     quote! { (|| Some(prefix.a.b.c.as_mut()?.d)) () },
-            // ),
-            // (
-            //     "a.b.c.d",
-            //     ReferenceKind::Immutable,
-            //     quote! { &prefix.a.b.c.d },
-            // ),
-            // (
-            //     "a.b.c.d",
-            //     ReferenceKind::Mutable,
-            //     quote! { &mut prefix.a.b.c.d },
-            // ),
-            // (
-            //     "a.b?.c?.d",
-            //     ReferenceKind::Immutable,
-            //     quote! { (|| Some(prefix.a.b.as_ref()?.c.as_ref()?.d)) () },
-            // ),
-            // (
-            //     "a.b?.c?.d",
-            //     ReferenceKind::Mutable,
-            //     quote! { (|| Some(prefix.a.b.as_mut()?.c.as_mut()?.d)) () },
-            // ),
-            // (
-            //     "a.b?.c?.d?",
-            //     ReferenceKind::Immutable,
-            //     quote! { (|| Some(prefix.a.b.as_ref()?.c.as_ref()?.d.as_ref()?)) () },
-            // ),
-            // (
-            //     "a.b?.c?.d?",
-            //     ReferenceKind::Mutable,
-            //     quote! { (|| Some(prefix.a.b.as_mut()?.c.as_mut()?.d.as_mut()?)) () },
-            // ),
-            // (
-            //     "a.b.c.d.e.f?.g.i.j.k.l.m.n",
-            //     ReferenceKind::Immutable,
-            //     quote! { (|| Some(prefix.a.b.c.d.e.f.as_ref()?.g.i.j.k.l.m.n)) () },
-            // ),
-            // (
-            //     "a.b.c.d.e.f?.g.i.j.k.l.m.n",
-            //     ReferenceKind::Mutable,
-            //     quote! { (|| Some(prefix.a.b.c.d.e.f.as_mut()?.g.i.j.k.l.m.n)) () },
-            // ),
-            // (
-            //     "a.b.c.d.e.f?.g.i.j.k.l.m.n?",
-            //     ReferenceKind::Immutable,
-            //     quote! { (|| Some(prefix.a.b.c.d.e.f.as_ref()?.g.i.j.k.l.m.n.as_ref()?)) () },
-            // ),
-            // (
-            //     "a.b.c.d.e.f?.g.i.j.k.l.m.n?",
-            //     ReferenceKind::Mutable,
-            //     quote! { (|| Some(prefix.a.b.c.d.e.f.as_mut()?.g.i.j.k.l.m.n.as_mut()?)) () },
+            //     quote! { match instance { CyclerInstance::InstanceA => &mut prefix.a.instance_a, CyclerInstance::InstanceB => &mut prefix.a.instance_b, } },
             // ),
         ];
         let cycler = Cycler {

--- a/crates/code_generation/src/accessor.rs
+++ b/crates/code_generation/src/accessor.rs
@@ -117,18 +117,6 @@ mod tests {
 
     #[test]
     fn paths_with_optionals_result_in_correct_accessor_token_streams() {
-        // "a?.b"
-        // "a?.b?"
-        // "a?.b?.c"
-        // "a?.b.c?"
-        // "a?.b.c?.d"
-        //
-        // "xxx.a?.b"
-        // "xxx.a?.b?"
-        // "xxx.a?.b?.c"
-        // "xxx.a?.b.c?"
-        // "xxx.a?.b.c?.d"
-
         let cases = [
             (
                 "a?.b",
@@ -201,21 +189,21 @@ mod tests {
                 ReferenceKind::Immutable,
                 quote! { (|| Some(&(*prefix.a.as_ref()?))) () },
             ),
-            // (
-            //     "$cycler_instance?",
-            //     ReferenceKind::Immutable,
-            //     quote! { match instance { CyclerInstance::InstanceA => (|| Some(prefix.instance_a.as_ref()?)) (), CyclerInstance::InstanceB => (|| Some(prefix.instance_b.as_ref()?)) (), } },
-            // ),
+            (
+                "$cycler_instance?",
+                ReferenceKind::Immutable,
+                quote! { match instance { CyclerInstance::InstanceA => (|| Some(&(*prefix.instance_a.as_ref()?))) (), CyclerInstance::InstanceB => (|| Some(&(*prefix.instance_b.as_ref()?))) (), } },
+            ),
             (
                 "a?",
                 ReferenceKind::Mutable,
                 quote! { (|| Some(&mut(*prefix.a.as_mut()?))) () },
             ),
-            // (
-            //     "$cycler_instance?",
-            //     ReferenceKind::Mutable,
-            //     quote! { match instance { CyclerInstance::InstanceA => (|| Some(prefix.instance_a.as_mut()?)) (), CyclerInstance::InstanceB => (|| Some(prefix.instance_b.as_mut()?)) (), } },
-            // ),
+            (
+                "$cycler_instance?",
+                ReferenceKind::Mutable,
+                quote! { match instance { CyclerInstance::InstanceA => (|| Some(&mut(*prefix.instance_a.as_mut()?))) (), CyclerInstance::InstanceB => (|| Some(&mut(*prefix.instance_b.as_mut()?))) (), } },
+            ),
             (
                 "a?.b?.c",
                 ReferenceKind::Immutable,
@@ -337,29 +325,19 @@ mod tests {
                 quote! { (|| Some(&mut(*&mut(*prefix.a.b.c.d.e.f.as_mut()?).g.i.j.k.l.m.n.as_mut()?))) () },
             ),
             ("a", ReferenceKind::Immutable, quote! { &prefix.a }),
-            // (
-            //     "$cycler_instance",
-            //     ReferenceKind::Immutable,
-            //     quote! { match instance { CyclerInstance::InstanceA => &prefix.instance_a, CyclerInstance::InstanceB => &prefix.instance_b, } },
-            // ),
             ("a", ReferenceKind::Mutable, quote! { &mut prefix.a }),
-            // (
-            //     "$cycler_instance",
-            //     ReferenceKind::Mutable,
-            //     quote! { match instance { CyclerInstance::InstanceA => &mut prefix.instance_a, CyclerInstance::InstanceB => &mut prefix.instance_b, } },
-            // ),
             ("a.b", ReferenceKind::Immutable, quote! { &prefix.a.b }),
-            // (
-            //     "a.$cycler_instance",
-            //     ReferenceKind::Immutable,
-            //     quote! { match instance { CyclerInstance::InstanceA => &prefix.a.instance_a, CyclerInstance::InstanceB => &prefix.a.instance_b, } },
-            // ),
+            (
+                "a.$cycler_instance",
+                ReferenceKind::Immutable,
+                quote! { match instance { CyclerInstance::InstanceA => &prefix.a.instance_a, CyclerInstance::InstanceB => &prefix.a.instance_b, } },
+            ),
             ("a.b", ReferenceKind::Mutable, quote! { &mut prefix.a.b }),
-            // (
-            //     "a.$cycler_instance",
-            //     ReferenceKind::Mutable,
-            //     quote! { match instance { CyclerInstance::InstanceA => &mut prefix.a.instance_a, CyclerInstance::InstanceB => &mut prefix.a.instance_b, } },
-            // ),
+            (
+                "a.$cycler_instance",
+                ReferenceKind::Mutable,
+                quote! { match instance { CyclerInstance::InstanceA => &mut prefix.a.instance_a, CyclerInstance::InstanceB => &mut prefix.a.instance_b, } },
+            ),
         ];
         let cycler = Cycler {
             name: "TestCycler".to_string(),

--- a/crates/code_generation/src/accessor.rs
+++ b/crates/code_generation/src/accessor.rs
@@ -1,4 +1,5 @@
 use convert_case::{Case, Casing};
+use itertools::{EitherOrBoth, Itertools};
 use proc_macro2::TokenStream;
 use quote::{format_ident, quote};
 use source_analyzer::{
@@ -76,23 +77,30 @@ fn path_to_accessor_token_stream_with_cycler_instance(
     };
 
     if path.contains_optional() {
-        let first_segment = segments.next().unwrap();
-        let x = segments.zip(path.segments.iter()).fold(
-            quote! {#prefix. #first_segment},
-            |acc, (token_stream, previous_segment)| match previous_segment.is_optional {
-                true => quote! {#reference (*#acc).#token_stream},
-                false => quote! {#acc.#token_stream},
-            },
-        );
+        let first_segment = segments
+            .next()
+            .expect("Path should always contain at least one segment");
 
-        match path.segments.last().unwrap().is_optional {
-            true => quote! {
-                (|| Some(#reference (*#x)))()
+        segments.zip_longest(path.segments.iter()).fold(
+            quote! {#prefix. #first_segment},
+            |recursive_token_stream, token_stream_segment_pair| match token_stream_segment_pair {
+                EitherOrBoth::Both(token_stream, previous_segment) => {
+                    match previous_segment.is_optional {
+                        true => quote! {#reference (*#recursive_token_stream).#token_stream},
+                        false => quote! {#recursive_token_stream.#token_stream},
+                    }
+                }
+                EitherOrBoth::Right(previous_segment) => match previous_segment.is_optional {
+                    true => quote! {
+                        (|| Some(#reference (*#recursive_token_stream)))()
+                    },
+                    false => quote! {
+                        (|| Some(#recursive_token_stream))()
+                    },
+                },
+                EitherOrBoth::Left(_) => unimplemented!(),
             },
-            false => quote! {
-                (|| Some(#x))()
-            },
-        }
+        )
     } else {
         quote! {
             #reference #prefix . #(#segments).*

--- a/crates/code_generation/src/accessor.rs
+++ b/crates/code_generation/src/accessor.rs
@@ -149,6 +149,16 @@ mod tests {
                 quote! { (|| Some(prefix.a.as_mut()?)) () },
             ),
             (
+                "a?.b",
+                ReferenceKind::Mutable,
+                quote! { (|| Some(prefix.a.as_mut()?.b)) () },
+            ),
+            (
+                "a?.b",
+                ReferenceKind::Immutable,
+                quote! { (|| Some(prefix.a.as_ref()?.b)) () },
+            ),
+            (
                 "$cycler_instance?",
                 ReferenceKind::Mutable,
                 quote! { match instance { CyclerInstance::InstanceA => (|| Some(prefix.instance_a.as_mut()?)) (), CyclerInstance::InstanceB => (|| Some(prefix.instance_b.as_mut()?)) (), } },

--- a/crates/code_generation/src/accessor.rs
+++ b/crates/code_generation/src/accessor.rs
@@ -79,7 +79,7 @@ fn path_to_accessor_token_stream_with_cycler_instance(
     if path.contains_optional() {
         let first_segment = segments
             .next()
-            .expect("Path should always contain at least one segment");
+            .expect("Path must always contain at least one segment");
 
         segments.zip_longest(path.segments.iter()).fold(
             quote! {#prefix. #first_segment},
@@ -98,7 +98,9 @@ fn path_to_accessor_token_stream_with_cycler_instance(
                         (|| Some(#recursive_token_stream))()
                     },
                 },
-                EitherOrBoth::Left(_) => unimplemented!(),
+                EitherOrBoth::Left(_) => {
+                    panic!("More segments than previous segments given. This should be impossible.")
+                }
             },
         )
     } else {

--- a/crates/code_generation/src/accessor.rs
+++ b/crates/code_generation/src/accessor.rs
@@ -83,11 +83,13 @@ fn path_to_accessor_token_stream_with_cycler_instance(
 
         segments.zip_longest(path.segments.iter()).fold(
             quote! {#prefix. #first_segment},
-            |recursive_token_stream, token_stream_segment_pair| match token_stream_segment_pair {
-                EitherOrBoth::Both(token_stream, previous_segment) => {
+            |recursive_token_stream, segment_pairs| match segment_pairs {
+                EitherOrBoth::Both(current_segment_token, previous_segment) => {
                     match previous_segment.is_optional {
-                        true => quote! {#reference (*#recursive_token_stream).#token_stream},
-                        false => quote! {#recursive_token_stream.#token_stream},
+                        true => {
+                            quote! {#reference (*#recursive_token_stream).#current_segment_token}
+                        }
+                        false => quote! {#recursive_token_stream.#current_segment_token},
                     }
                 }
                 EitherOrBoth::Right(previous_segment) => match previous_segment.is_optional {

--- a/crates/code_generation/src/cyclers.rs
+++ b/crates/code_generation/src/cyclers.rs
@@ -44,7 +44,7 @@ fn generate_module(cycler: &Cycler, cyclers: &Cyclers) -> TokenStream {
     let cycler_implementation = generate_implementation(cycler, cyclers);
 
     quote! {
-        #[allow(dead_code, unused_mut, unused_variables, clippy::too_many_arguments, clippy::needless_question_mark)]
+        #[allow(dead_code, unused_mut, unused_variables, clippy::too_many_arguments, clippy::needless_question_mark, clippy::borrow_deref_ref)]
         pub(crate) mod #module_name {
             use color_eyre::eyre::WrapErr;
             use crate::structs::#module_name::{MainOutputs, AdditionalOutputs};

